modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APISecurityTestCase.java

```diff
@@ -455,6 +456,115 @@ public void testAPIInvocationWithMutualSSLHeader() throws IOException, XPathExpr
                 "Mutual SSL Authentication has succeeded for a different certificate");
     }
 
+    @Test(description = "Testing the invocation with API Keys having IP restriction",
+            dependsOnMethods = {"testCreateAndPublishAPIWithOAuth2"})
+    public void testInvocationWithApiKeysWithIPCondition() throws Exception {
+        String permittedIP = "152.23.5.6, 192.168.1.2/24, 2001:c00::/23";
+        APIKeyDTO apiKeyDTO = restAPIStore
+                .generateAPIKeys(applicationId, ApplicationKeyGenerateRequestDTO.KeyTypeEnum.PRODUCTION.toString(),
+                        -1, permittedIP, null);
+
+        assertNotNull(apiKeyDTO, "API Key generation failed");
+
+        Map<String, String> requestHeaders1 = new HashMap<>();
```

Comment:
Looks like code is repeating up to L512, Can't we move the requestHeaders preparation to separate method and keep only assert check here?

Reply:
Fixed in https://github.com/wso2/product-apim/pull/9200